### PR TITLE
Issue fixed for bigint, numeric and decimal data types

### PR DESCRIPTION
**Pull request by deepakshi-mittal**
Task: BABEL-4315

### Description
 
Added different functions to handle int, smallest and bigint input 
Added test cases covering join, all data types, temp tables 
Changed error message for unsupported datatypes(ie non-integer types) 
Numeric and decimal to be treated as biting, same as that of create table behavior with identity column

This issue is related to BABEL-539

### Issues Resolved

[List any issues this PR will resolve]

### Test Scenarios Covered ###
* **Use case based -**

* **Boundary conditions -**

* **Arbitrary inputs -**

* **Negative test cases -**

* **Minor version upgrade tests -**

* **Major version upgrade tests -**

* **Performance tests -**

* **Tooling impact -**

* **Client tests -**

### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/src/pl_handler.c [hunk 1]
```diff
@@ -5070,6 +5071,39 @@ static char *get_oid_type_string(int type_oid){
...
+			fxpr = (FuncExpr *)node;
+			foreach(lc, fxpr->args)
+			{
+				Node *n = (Node *)lfirst(lc);
+				if (IsA(n, Const))
+				{
+					con = (Const *)n;
+					val = (int64) DatumGetInt64(con->constvalue);
+					break;
+				}
```
Comment (forestkeeper) [inline contrib/babelfishpg_tsql/src/pl_handler.c:5097]: add an else in here and throw out some error

File: contrib/babelfishpg_tsql/src/pl_handler.c [hunk 2]
```diff
@@ -5094,19 +5128,19 @@ static List *transformSelectIntoStmt(CreateTableAsStmt *stmt, const char *queryS
 		{
 			TargetEntry *tle = (TargetEntry *)lfirst(elements);
 
-			if (tle->expr && IsA(tle->expr, FuncExpr) && strcasecmp(get_func_name(((FuncExpr *)(tle->expr))->funcid), "identity_into") == 0)
+			if (tle->expr && IsA(tle->expr, FuncExpr) && strncmp(get_func_name(((FuncExpr *)(tle->expr))->funcid), "identity_into", strlen( "identity_into")) == 0)
```
Comment (forestkeeper) [inline contrib/babelfishpg_tsql/src/pl_handler.c:5131]: Would you mind add another test case using IDENTITY in upper case ?

File: contrib/babelfishpg_tsql/runtime/functions.c [hunk 3]
```diff
@@ -1898,14 +1901,32 @@ bbf_set_context_info(PG_FUNCTION_ARGS)
...
 	PG_RETURN_VOID();
 }
-
-Datum
-identity_into(PG_FUNCTION_ARGS)
-{
+/**
+ * Common function to get next value of sequence for Identity function in Select Into
+*/
+static int64 get_identity_next_value(void){
```
Comment (forestkeeper) [inline contrib/babelfishpg_tsql/runtime/functions.c:1907]: the '{' should be in the next line

**Comment by deepakshi-mittal** [comment]
Closing this Pr because of Merge issues

File: contrib/babelfishpg_tsql/src/pl_handler.c (hunk 1 above)
Comment (deepakshi-mittal) [reply contrib/babelfishpg_tsql/src/pl_handler.c:5097]: okay

**Pull request closed without merging**